### DISCUSSION TITLE
how to install java sdk with maven?

### DISCUSSION THREAD

**Discussion by sanqiuli** [argoproj/argo-workflows#9194] [Q&A] [answered]
I read the md documentation in https://github.com/argoproj/argo-workflows/tree/master/sdks/java this link, I have never used github's maven repository, so I don't know how to use maven argo-client-java is imported into my java project, can someone tell me how to import the jar package of github's maven repository, thank you! ! !

**Comment by terrytangyuan** [comment, marked as answer]
Try this:
```xml
    <profile>
      <id>github</id>
      <repositories>
        <repository>
          <id>central</id>
          <url>https://repo1.maven.org/maven2</url>
          <releases><enabled>true</enabled></releases>
          <snapshots><enabled>false</enabled></snapshots>
        </repository>
        <repository>
          <id>github</id>
          <name>GitHub argoproj Apache Maven Packages</name>
          <url>https://maven.pkg.github.com/argoproj/argo-workflows</url>
        </repository>
      </repositories>
    </profile>
```

**Reply by sanqiuli** [reply]
add this to pom.xml or setting.xml?

**Reply by terrytangyuan** [reply]
settings.xml

**Reply by terrytangyuan** [reply]
See https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry

**Reply by sanqiuli** [reply]
it works，thanks！

**Comment by masalinas** [comment]
I followed the recomendation from @terrytangyuan in my maven settings.xml and add the depedency in my pom file like this:

```xml
		<dependency>
		    <groupId>io.argoproj.workflow</groupId>
		    <artifactId>argo-client-java</artifactId>
		    <version>v3.3.8</version>
		</dependency>
```

Also using this dependency
```xml
		<dependency>
		    <groupId>io.argoproj.workflow</groupId>
		    <artifactId>argo-client-java</artifactId>
		    <version>v3.4.5</version>
		</dependency>
```

In any test it not works. My Eclipse UI detects it as missing artifact. What is it wrong?
